File: core/src/ua/syt0r/GameManager.java
package ua.syt0r;

import com.badlogic.gdx.Game;
import com.badlogic.gdx.graphics.FPSLogger;
import ua.syt0r.screens.main.MainMenuScreen;

public class GameManager extends Game {

    private FPSLogger fpsLogger;

    @Override
    public void create() {

        Assets.loadLoadingScreenAssets();

        ScreenManager.getInstance().initialize(this);
        ScreenManager.getInstance().showScreen(new MainMenuScreen());

        fpsLogger = new FPSLogger();
    }

    @Override
    public void render() {
        super.render();
        fpsLogger.log();
    }

    @Override
    public void resize(int width, int height) {
        super.resize(width, height);
    }

    @Override
    public void dispose() {
        super.dispose();
        Assets.dispose();
    }

    @Override
    public void pause() {
        super.pause();
    }

}

File: core/src/ua/syt0r/GameManager.kt
package ua.syt0r

import com.badlogic.gdx.Game
import com.badlogic.gdx.Screen
import com.badlogic.gdx.graphics.FPSLogger
import ua.syt0r.Assets.loadLoadingScreenAssets
import ua.syt0r.screens.main.MainMenuScreen

class GameManager(
    private val initialScreenCreator: () -> Screen = { MainMenuScreen() }
) : Game() {

    private lateinit var fpsLogger: FPSLogger

    override fun create() {
        loadLoadingScreenAssets()
        ScreenManager.getInstance().initialize(this)
        ScreenManager.getInstance().showScreen(initialScreenCreator())
        fpsLogger = FPSLogger()
    }

    override fun render() {
        super.render()
        fpsLogger.log()
    }

    override fun dispose() {
        super.dispose()
        Assets.dispose()
    }

}
